### PR TITLE
Extension for Traffic Sign classification for better harmonisation with OpenDRIVE

### PR DESCRIPTION
**Pull request by MarcelOliver**
#### Add a description
OpenDRIVE categorization for traffic signs is more generic than OSI enum. 
[OpenDRIVE 1.6](https://releases.asam.net/OpenDRIVE/1.6.0/ASAM_OpenDRIVE_BS_V1-6-0.html#_signals) definition for traffic sign:
2021-02-11_15h02_03

**Some questions to ask**:
What is this change?
Extension for traffic sign classification (mainSign and sulpplementarySign). Adding attributes **country**, **country_revision**, **code** and **sub_code** in order to provide a better traffic sign harmonisation with OpenDRIVE.

2021-02-11_13h50_04

#### Mention a member
@HendrikAmelunxen @pmai 

#### Check the checklist

- [ ] My code and comments follow the [style guidelines](https://opensimulationinterface.github.io/osi-documentation/osi/commenting.html) and [contributors guidelines](https://opensimulationinterface.github.io/osi-documentation/osi/howtocontribute.html) of this project.
- [ ] I have performed a self-review of my own code.
- [ ] I have made corresponding changes to the [documentation](https://github.com/OpenSimulationInterface/osi-documentation).
- [ ] My changes generate no new warnings.
- [ ] I have added tests that prove my fix is effective or that my feature works.
- [ ] New and existing unit tests / travis ci pass locally with my changes.

### CODE REVIEW

**Review by HendrikAmelunxen — APPROVED** [review]
Approved; as discussed in OSI Harmonization working group

**Comment by stefancyliax** [comment]
OSI CCB:
- verify the proto comments (styling) (Pierre)
- documentation review: how should we mark deprecated parts of the code (Max)

**Comment by kmeids** [comment]
**Output from CCB 26.05.2021**
1. @max-rosin please do a documentation review and set to ReadyForCCBReview when done.

File: osi_roadmarking.proto [hunk 1]
```diff
@@ -121,15 +121,43 @@ message RoadMarking
 
 
         // Boolean flag to indicate that the road marking is taken out of service.
-        // This can be achieved by visibly crossing the road marking with stripes, 
-        // or completly covering a road marking making it not visible. 
+        // This can be achieved by visibly crossing the road marking with stripes,
+        // or completly covering a road marking making it not visible.
```
Comment (max-rosin) [inline osi_roadmarking.proto:125]: Typo: "completely"...

File: osi_roadmarking.proto [hunk 2]
```diff
@@ -121,15 +121,43 @@ message RoadMarking
...
+        // sub_code is only unique in combination with #country and #country_revision.  <br>
+        // e.g. http://www.vzkat.de/2017/VzKat.htm
+        //
+        optional string sub_code = 11;
+
         // Definition of road marking types.
         //
+        // \attention Deprecated: OpenDRIVE categorization for traffic signs is more generic than OSI enum
+        // (see <a href="https://releases.asam.net/OpenDRIVE/1.6.0/ASAM_OpenDRIVE_BS_V1-6-0.html#_signals">OpenDRIVE V1-6-0</a>).<br>
+        // How the Type will be dealt in future will be discussed in Harmonization OpenX group.<br>
```
Comment (max-rosin) [inline osi_roadmarking.proto:158]: Typo: "dealt with..."

File: osi_roadmarking.proto [hunk 3]
```diff
@@ -121,15 +121,43 @@ message RoadMarking
...
+        // Deprecated planned for version 4.0.0
```
Comment (max-rosin) [inline osi_roadmarking.proto:159]: Better: "Deprecation planned for..." or "Deprecated status planned for..."

File: osi_trafficsign.proto [hunk 4]
```diff
@@ -225,6 +225,10 @@ message TrafficSign
 
             // The type of the traffic sign.
             //
+            // \attention Deprecated: OpenDRIVE categorization for traffic signs is more generic than OSI enum
+            // (see <a href="https://releases.asam.net/OpenDRIVE/1.6.0/ASAM_OpenDRIVE_BS_V1-6-0.html#_signals">OpenDRIVE V1-6-0</a>).<br>
+            // How the Type type will be dealt in future will be discussed in Harmonization OpenX group.<br>
+            // Deprecated planned for version 4.0.0
```
Comment (max-rosin) [inline osi_trafficsign.proto:231]: Better: "Deprecation planned for..." or "Deprecated status planned for..."

File: osi_trafficsign.proto [hunk 5]
```diff
@@ -225,6 +225,10 @@ message TrafficSign
 
             // The type of the traffic sign.
             //
+            // \attention Deprecated: OpenDRIVE categorization for traffic signs is more generic than OSI enum
+            // (see <a href="https://releases.asam.net/OpenDRIVE/1.6.0/ASAM_OpenDRIVE_BS_V1-6-0.html#_signals">OpenDRIVE V1-6-0</a>).<br>
+            // How the Type type will be dealt in future will be discussed in Harmonization OpenX group.<br>
```
Comment (max-rosin) [inline osi_trafficsign.proto:230]: Typo: "dealt with..."

File: osi_trafficsign.proto [hunk 6]
```diff
@@ -306,6 +341,10 @@ message TrafficSign
             // https://www.gesetze-im-internet.de/stvo_2013/anlage_4.html
             // (Verkehrseinrichtungen) \arg https://traffic-rules.com/
             //
+            // \attention Deprecated: OpenDRIVE categorization for traffic signs is more generic than OSI enum
+            // (see <a href="https://releases.asam.net/OpenDRIVE/1.6.0/ASAM_OpenDRIVE_BS_V1-6-0.html#_signals">OpenDRIVE V1-6-0</a>).<br>
+            // How the Type will be dealt in future will be discussed in Harmonization OpenX group.<br>
+            // Deprecated planned for version 4.0.0
```
Comment (max-rosin) [inline osi_trafficsign.proto:347]: Better: "Deprecation planned for..." or "Deprecated status planned for..."

File: osi_trafficsign.proto [hunk 7]
```diff
@@ -5498,6 +5537,10 @@ message TrafficSign
 
             // Type of the supplementary sign.
             //
+            // \attention Deprecated: OpenDRIVE categorization for traffic signs is more generic than OSI enum
+            // (see <a href="https://releases.asam.net/OpenDRIVE/1.6.0/ASAM_OpenDRIVE_BS_V1-6-0.html#_signals">OpenDRIVE V1-6-0</a>).<br>
+            // How the Type type will be dealt in future will be discussed in Harmonization OpenX group.<br>
+            // Deprecated planned for version 4.0.0
```
Comment (max-rosin) [inline osi_trafficsign.proto:5543]: Better: "Deprecation planned for..." or "Deprecated status planned for..."

File: osi_trafficsign.proto [hunk 8]
```diff
@@ -5527,17 +5570,48 @@ message TrafficSign
             // bikes, cars, trucks and so on), that the supplementary sign
             // makes reference to.
             //
+            // \attention Deprecated: OpenDRIVE categorization for traffic signs is more generic than OSI enum
+            // (see <a href="https://releases.asam.net/OpenDRIVE/1.6.0/ASAM_OpenDRIVE_BS_V1-6-0.html#_signals">OpenDRIVE V1-6-0</a>).<br>
+            // How the Actor actor will be dealt in future will be discussed in Harmonization OpenX group.<br>
+            // Deprecated planned for version 4.0.0
```
Comment (max-rosin) [inline osi_trafficsign.proto:5576]: Better: "Deprecation planned for..." or "Deprecated status planned for..."

File: osi_trafficsign.proto [hunk 9]
```diff
@@ -5527,17 +5570,48 @@ message TrafficSign
...
             repeated Actor actor = 5;
 
             // A direction arrow shown on the supplementary sign.
             //
+            // \attention Deprecated: OpenDRIVE categorization for traffic signs is more generic than OSI enum
+            // (see <a href="https://releases.asam.net/OpenDRIVE/1.6.0/ASAM_OpenDRIVE_BS_V1-6-0.html#_signals">OpenDRIVE V1-6-0</a>).<br>
+            // How the Arrow arrow will be dealt in future will be discussed in Harmonization OpenX group.<br>
+            // Deprecated planned for version 4.0.0
```
Comment (max-rosin) [inline osi_trafficsign.proto:5584]: Better: "Deprecation planned for..." or "Deprecated status planned for..."

File: osi_trafficsign.proto [hunk 10]
```diff
@@ -5551,6 +5625,11 @@ message TrafficSign
             // text, \c Type is used in descending order in the following
             // sequence: \c #TYPE_EXCEPT, \c #TYPE_CONSTRAINED_TO, \c
             // #TYPE_ARROW, \c #TYPE_TIME, \c #TYPE_SPACE, \c #TYPE_TEXT.
+            //
+            // \attention Deprecated: OpenDRIVE categorization for traffic signs is more generic than OSI enum
+            // (see <a href="https://releases.asam.net/OpenDRIVE/1.6.0/ASAM_OpenDRIVE_BS_V1-6-0.html#_signals">OpenDRIVE V1-6-0</a>).<br>
+            // How the enum Type will be dealt in future will be discussed in Harmonization OpenX group.<br>
+            // Deprecated planned for version 4.0.0
```
Comment (max-rosin) [inline osi_trafficsign.proto:5632]: Better: "Deprecation planned for..." or "Deprecated status planned for..."

File: osi_trafficsign.proto [hunk 11]
```diff
@@ -8966,6 +9045,10 @@ message TrafficSign
             // Definition of the traffic actors the supplementary sign makes
             // reference to. E.g. bikes, trucks, cars, etc.
             //
+            // \attention Deprecated: OpenDRIVE categorization for traffic signs is more generic than OSI enum
+            // (see <a href="https://releases.asam.net/OpenDRIVE/1.6.0/ASAM_OpenDRIVE_BS_V1-6-0.html#_signals">OpenDRIVE V1-6-0</a>).<br>
+            // How the enum Actor will be dealt in future will be discussed in Harmonization OpenX group.<br>
+            // Deprecated planned for version 4.0.0
```
Comment (max-rosin) [inline osi_trafficsign.proto:9051]: Better: "Deprecation planned for..." or "Deprecated status planned for..."

File: osi_trafficsign.proto [hunk 12]
```diff
@@ -9191,6 +9274,10 @@ message TrafficSign
 
                 // The direction to which the arrow points to
                 //
+                // \attention Deprecated: OpenDRIVE categorization for traffic signs is more generic than OSI enum
+                // (see <a href="https://releases.asam.net/OpenDRIVE/1.6.0/ASAM_OpenDRIVE_BS_V1-6-0.html#_signals">OpenDRIVE V1-6-0</a>).<br>
+                // How the enum Direction will be dealt in future will be discussed in Harmonization OpenX group.<br>
+                // Deprecated planned for version 4.0.0
```
Comment (max-rosin) [inline osi_trafficsign.proto:9280]: Better: "Deprecation planned for..." or "Deprecated status planned for..."

File: osi_trafficsign.proto [hunk 13]
```diff
@@ -9191,6 +9274,10 @@ message TrafficSign
 
                 // The direction to which the arrow points to
                 //
+                // \attention Deprecated: OpenDRIVE categorization for traffic signs is more generic than OSI enum
+                // (see <a href="https://releases.asam.net/OpenDRIVE/1.6.0/ASAM_OpenDRIVE_BS_V1-6-0.html#_signals">OpenDRIVE V1-6-0</a>).<br>
+                // How the enum Direction will be dealt in future will be discussed in Harmonization OpenX group.<br>
```
Comment (max-rosin) [inline osi_trafficsign.proto:9279]: Typo: "dealt with..."

File: osi_trafficsign.proto [hunk 14]
```diff
@@ -8966,6 +9045,10 @@ message TrafficSign
             // Definition of the traffic actors the supplementary sign makes
             // reference to. E.g. bikes, trucks, cars, etc.
             //
+            // \attention Deprecated: OpenDRIVE categorization for traffic signs is more generic than OSI enum
+            // (see <a href="https://releases.asam.net/OpenDRIVE/1.6.0/ASAM_OpenDRIVE_BS_V1-6-0.html#_signals">OpenDRIVE V1-6-0</a>).<br>
+            // How the enum Actor will be dealt in future will be discussed in Harmonization OpenX group.<br>
```
Comment (max-rosin) [inline osi_trafficsign.proto:9050]: Typo: "dealt with..."

File: osi_trafficsign.proto [hunk 15]
```diff
@@ -5551,6 +5625,11 @@ message TrafficSign
             // text, \c Type is used in descending order in the following
             // sequence: \c #TYPE_EXCEPT, \c #TYPE_CONSTRAINED_TO, \c
             // #TYPE_ARROW, \c #TYPE_TIME, \c #TYPE_SPACE, \c #TYPE_TEXT.
+            //
+            // \attention Deprecated: OpenDRIVE categorization for traffic signs is more generic than OSI enum
+            // (see <a href="https://releases.asam.net/OpenDRIVE/1.6.0/ASAM_OpenDRIVE_BS_V1-6-0.html#_signals">OpenDRIVE V1-6-0</a>).<br>
+            // How the enum Type will be dealt in future will be discussed in Harmonization OpenX group.<br>
```
Comment (max-rosin) [inline osi_trafficsign.proto:5631]: Typo: "dealt with..."

File: osi_trafficsign.proto [hunk 16]
```diff
@@ -5527,17 +5570,48 @@ message TrafficSign
...
+            // (see <a href="https://releases.asam.net/OpenDRIVE/1.6.0/ASAM_OpenDRIVE_BS_V1-6-0.html#_signals">OpenDRIVE V1-6-0</a>).<br>
+            // How the Actor actor will be dealt in future will be discussed in Harmonization OpenX group.<br>
+            // Deprecated planned for version 4.0.0
             repeated Actor actor = 5;
 
             // A direction arrow shown on the supplementary sign.
             //
+            // \attention Deprecated: OpenDRIVE categorization for traffic signs is more generic than OSI enum
+            // (see <a href="https://releases.asam.net/OpenDRIVE/1.6.0/ASAM_OpenDRIVE_BS_V1-6-0.html#_signals">OpenDRIVE V1-6-0</a>).<br>
+            // How the Arrow arrow will be dealt in future will be discussed in Harmonization OpenX group.<br>
```
Comment (max-rosin) [inline osi_trafficsign.proto:5583]: Typo: "dealt with..."

File: osi_trafficsign.proto [hunk 17]
```diff
@@ -5527,17 +5570,48 @@ message TrafficSign
             // bikes, cars, trucks and so on), that the supplementary sign
             // makes reference to.
             //
+            // \attention Deprecated: OpenDRIVE categorization for traffic signs is more generic than OSI enum
+            // (see <a href="https://releases.asam.net/OpenDRIVE/1.6.0/ASAM_OpenDRIVE_BS_V1-6-0.html#_signals">OpenDRIVE V1-6-0</a>).<br>
+            // How the Actor actor will be dealt in future will be discussed in Harmonization OpenX group.<br>
```
Comment (max-rosin) [inline osi_trafficsign.proto:5575]: Typo: "dealt with..."

File: osi_trafficsign.proto [hunk 18]
```diff
@@ -5498,6 +5537,10 @@ message TrafficSign
 
             // Type of the supplementary sign.
             //
+            // \attention Deprecated: OpenDRIVE categorization for traffic signs is more generic than OSI enum
+            // (see <a href="https://releases.asam.net/OpenDRIVE/1.6.0/ASAM_OpenDRIVE_BS_V1-6-0.html#_signals">OpenDRIVE V1-6-0</a>).<br>
+            // How the Type type will be dealt in future will be discussed in Harmonization OpenX group.<br>
```
Comment (max-rosin) [inline osi_trafficsign.proto:5542]: Typo: "dealt with..."

File: osi_trafficsign.proto [hunk 19]
```diff
@@ -306,6 +341,10 @@ message TrafficSign
             // https://www.gesetze-im-internet.de/stvo_2013/anlage_4.html
             // (Verkehrseinrichtungen) \arg https://traffic-rules.com/
             //
+            // \attention Deprecated: OpenDRIVE categorization for traffic signs is more generic than OSI enum
+            // (see <a href="https://releases.asam.net/OpenDRIVE/1.6.0/ASAM_OpenDRIVE_BS_V1-6-0.html#_signals">OpenDRIVE V1-6-0</a>).<br>
+            // How the Type will be dealt in future will be discussed in Harmonization OpenX group.<br>
```
Comment (max-rosin) [inline osi_trafficsign.proto:346]: Typo: "dealt with..."

File: osi_trafficsign.proto [hunk 20]
```diff
@@ -280,12 +288,39 @@ message TrafficSign
             // As for every boolean in the protocol buffers language, the
             // default value of \c #vertically_mirrored is \c false.
             //
+            // \attention Deprecated: OpenDRIVE categorization for traffic signs is more generic than OSI enum
+            // (see <a href="https://releases.asam.net/OpenDRIVE/1.6.0/ASAM_OpenDRIVE_BS_V1-6-0.html#_signals">OpenDRIVE V1-6-0</a>).<br>
+            // How the bool vertically_mirrored will be dealt in future will be discussed in Harmonization OpenX group.<br>
```
Comment (max-rosin) [inline osi_trafficsign.proto:293]: Typo: "dealt with..."

File: osi_trafficsign.proto [hunk 21]
```diff
@@ -256,6 +260,10 @@ message TrafficSign
             // might have been intentionally unmounted and, hence, not be in
             // effect.
             //
+            // \attention Deprecated: OpenDRIVE categorization for traffic signs is more generic than OSI enum
+            // (see <a href="https://releases.asam.net/OpenDRIVE/1.6.0/ASAM_OpenDRIVE_BS_V1-6-0.html#_signals">OpenDRIVE V1-6-0</a>).<br>
+            // How the DirectionScope direction_scope will be dealt in future will be discussed in Harmonization OpenX group.<br>
```
Comment (max-rosin) [inline osi_trafficsign.proto:265]: Typo: "dealt with..."

File: osi_trafficsign.proto [hunk 22]
```diff
@@ -256,6 +260,10 @@ message TrafficSign
...
+            // Deprecated planned for version 4.0.0
```
Comment (max-rosin) [inline osi_trafficsign.proto:266]: Better: "Deprecation planned for..." or "Deprecated status planned for..."

File: osi_trafficsign.proto [hunk 23]
```diff
@@ -280,12 +288,39 @@ message TrafficSign
             // As for every boolean in the protocol buffers language, the
             // default value of \c #vertically_mirrored is \c false.
             //
+            // \attention Deprecated: OpenDRIVE categorization for traffic signs is more generic than OSI enum
+            // (see <a href="https://releases.asam.net/OpenDRIVE/1.6.0/ASAM_OpenDRIVE_BS_V1-6-0.html#_signals">OpenDRIVE V1-6-0</a>).<br>
+            // How the bool vertically_mirrored will be dealt in future will be discussed in Harmonization OpenX group.<br>
+            // Deprecated planned for version 4.0.0
```
Comment (max-rosin) [inline osi_trafficsign.proto:294]: Better: "Deprecation planned for..." or "Deprecated status planned for..."

**Review by max-rosin — CHANGES_REQUESTED** [review]

**Review by max-rosin — APPROVED** [review]

File: osi_roadmarking.proto [hunk 24]
```diff
@@ -121,15 +121,43 @@ message RoadMarking
...
+        // sub_code is only unique in combination with #country and #country_revision.  <br>
+        // e.g. http://www.vzkat.de/2017/VzKat.htm
+        //
+        optional string sub_code = 11;
+
         // Definition of road marking types.
         //
+        // \attention Deprecated: OpenDRIVE categorization for traffic signs is more generic than OSI enum
+        // (see <a href="https://releases.asam.net/OpenDRIVE/1.6.0/ASAM_OpenDRIVE_BS_V1-6-0.html#_signals">OpenDRIVE V1-6-0</a>).<br>
+        // How the Type will be dealt with in future will be discussed in Harmonization OpenX group.<br>
```
Comment (ThomasNaderBMW) [inline osi_roadmarking.proto:158]: in the future?

**Comment by stefancyliax** [comment]
CCB 13.07.2021:
Review in CCB:
- Harmonization topic for "type" in relation to OpenDRIVE. Please add name of corresponding field in OpenDRIVE.
- add general hint to OpenDRIVE
- "Deprecated status planned for version 4.0.0" needs to be changed to "revision planned for version 4.0.0"
to be done by @pmai

**Comment by kmeids** [comment]
**CCB Output 21.07.2021**
1. @pmai will be able to merge this.

**Pull request merged**